online-shop/src/app/modules/shared/shopping-cart/components/containers/shopping-cart-details/shopping-cart-details.component.ts

```diff
@@ -1,17 +1,37 @@
 import { Component } from '@angular/core';
-import { products } from 'src/app/mocks/products.mocks';
 import { Product } from 'src/app/modules/shared/types/products.types';
-import { ShoppingCartService } from 'src/app/services/shopping-cart-service';
-
+import { ShoppingCartService } from 'src/app/modules/shared/shopping-cart/services/shopping-cart-service';
+import { OrderCreateDto } from 'src/app/modules/shared/types/order-create-dto.types';
+import { HttpClient } from '@angular/common/http';
+import { environment } from 'src/environments/environment';
 @Component({
   selector: 'app-shopping-cart-details',
   templateUrl: './shopping-cart-details.component.html',
   styleUrls: ['./shopping-cart-details.component.scss'],
 })
 export class ShoppingCartDetailsComponent {
   products: Product[] = [];
-  constructor(private cartService:ShoppingCartService )
-  {
-    this.products=this.cartService.getCartItems();
+  apiUrl = environment.apiUrl;
+
+  constructor(
+    private cartService: ShoppingCartService,
+    private http: HttpClient
+  ) {
+    this.products = this.cartService.getCartItems();
```

Comment:
I would suggest to initialize data on ngOnInit (left a comment below with a link in [shopping-cart-service.ts](https://github.com/VladVeltanMSG/Angular-Training/pull/2/files#diff-63943789b35bf08eef77d622f7eadbcc0751a3a44a0d2673db589c5f94fe61a1))

Reply:
I now do it ngOnInit